relay-profiling/src/sample.rs

```diff
@@ -232,20 +232,27 @@ fn parse_profile(payload: &[u8]) -> Result<SampleProfile, ProfileError> {
         return Err(ProfileError::MissingProfileMetadata);
     }
 
-    let Some(transaction) = profile.transactions.get(0).cloned() else {
-        return Err(ProfileError::NoTransactionAssociated);
+    // Clean samples before running the checks.
+    profile.remove_single_samples_per_thread();
+
+    let transaction = match &profile.transaction {
+        Some(transaction) => transaction,
+        None => profile
+            .transactions
+            .get(0)
+            .ok_or(ProfileError::NoTransactionAssociated)?,
```

Comment:
You could `.take()` both `profile.transaction` and `profile.transactions`, then you would not have to clone further down.